### PR TITLE
Test blockly core in build

### PR DESCRIPTION
**Pull request by Bjvanminnen**

### CODE REVIEW

File: blockly-core/test.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+cd tests
+
+# Make sure phantomjs 2+ is installed
+phantomjsversion=$(node_modules/phantomjs/bin/phantomjs --version)
+if [ $phantomjsversion != "2.0.0" ]; then
+  npm install
```
Comment (Bjvanminnen) [inline blockly-core/test.sh]: I'm trying to decide whether I like the idea of running npm install here directly vs. outputting a message telling the user they should do that. It seems useful this way, but in our other projects there aren't any instances of doing something like this.

File: blockly-core/tests/phantomwrapper.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            typeof(onReady) === "string" ? eval(onReady) : onReady(); //< Do what it's supposed to do once the condition is fulfilled
+            clearInterval(interval); //< Stop this interval
+          }
+        }
+      }, 250); //< repeat check every 250ms
+};
+
+// This is a thin wrapper around using phantomjs to load and run the blockly core tests
+var fs = require('fs');
+var page = require('webpage').create();
```
Comment (Bjvanminnen) [inline blockly-core/tests/phantomwrapper.js]: I prefer seeing all requires at the top of the page

File: blockly-core/test.sh (hunk 1 above)
Comment (islemaster) [inline blockly-core/test.sh]: If we remove the automated npm install here, I must have to update some other script to make sure our CI environments npm install as needed, right?  Do you know where that is?

Comment (Bjvanminnen) [inline blockly-core/test.sh]: code-dot-org/Rakefile i think

File: blockly-core/tests/phantomwrapper.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      if (/\[FAILED\]/.test(page.plainText)) {
+        console.log(RED + "Blockly-Core tests failed" + DEFAULT_STYLE);
+        phantom.exit(1);
+        return;
+      }
+
+      console.log(GREEN + "Blockly-Core tests passed" + DEFAULT_STYLE);
+      phantom.exit(0);
+    });
+});
```
Comment (Bjvanminnen) [inline blockly-core/tests/phantomwrapper.js]: indenting too much

File: blockly-core/tests/phantomwrapper.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
+var fs = require('fs');
+var WebPage = require('webpage');
+
+var DEFAULT_STYLE = "\033[39m";
+var GREEN = "\033[32m";
+var RED = "\033[31m";
```
Comment (Bjvanminnen) [inline blockly-core/tests/phantomwrapper.js]: You could also use something like https://github.com/Marak/colors.js if you wanted.

**Comment by Bjvanminnen** [comment]
lgtm. good stuff!

File: blockly-core/tests/phantomwrapper.js (hunk 4 above)
Comment (islemaster) [inline blockly-core/tests/phantomwrapper.js]: Tried that actually.  It did color my output, but also generated warnings that I couldn't get rid of - something about us running the phantomjs executable directly, not using node the normal way. I figure plan B here is good enough for now.

Comment (Bjvanminnen) [inline blockly-core/tests/phantomwrapper.js]: Makes sense.

**Pull request merged**